packages/tools/src/tools/annotation/EllipticalROITool.ts

```diff
@@ -765,9 +766,22 @@ class EllipticalROITool extends AnnotationTool {
       const canvasCoordinates = points.map((p) =>
         viewport.worldToCanvas(p)
       ) as [Types.Point2, Types.Point2, Types.Point2, Types.Point2];
-      const canvasCorners = <Array<Types.Point2>>(
-        getCanvasEllipseCorners(canvasCoordinates)
-      );
+
+      const rotation = Math.abs(viewport.getRotation() - data.initialRotation);
+      let canvasCorners;
+
+      if (rotation == 90 || rotation == 270) {
```

Comment:
Can you check the save/restore in the adapters please?

Reply:
Is this what you are referring to?

https://github.com/cornerstonejs/cornerstone3D-beta/blob/b989d505098a2d1dcdcb23c9af7ace719f07dcbd/packages/adapters/src/adapters/Cornerstone3D/EllipticalROI.ts#L96

would you like me to follow this approach here too?

Reply:
No, I want you to run OHIF, create some markup with different initial rotations, then save it, and see if it still displays the same object after it loads back in.  You may need to do something special in the adapter to make sure it gets saved/restored correctly.

Reply:
Yeah it doesn’t save correctly, I’ll work on this and update you.

Reply:
It works as expected now, before saving, I convert so the rotation is 0, so when its restored we won't need the rotation value since it will always be 0.

FYI the DICOMSRDisplayTool in OHIF needs to change so it renders properly, only the cornerstone native tool renders properly now.

It renders fine in my GIF because I modified the DICOMSRDisplayTool locally to account for the rotation the same way I did it for the native cornerstone tool.

```JS
    const rotation = viewport.getRotation();

      canvasCoordinates = ellipsePointsWorld.map(p =>
        viewport.worldToCanvas(p)
      );
      let canvasCorners;
      if (rotation == 90 || rotation == 270) {
        canvasCorners = <Array<Types.Point2>>(
          utilities.math.ellipse.getCanvasEllipseCorners([
            canvasCoordinates[2],
            canvasCoordinates[3],
            canvasCoordinates[0],
            canvasCoordinates[1],
          ])
        );
      } else {
        canvasCorners = <Array<Types.Point2>>(
          utilities.math.ellipse.getCanvasEllipseCorners(canvasCoordinates)
        );
      }
```

![GIF 6](https://user-images.githubusercontent.com/93064150/225842654-6d19b0e3-978d-4a3e-ab5a-7a1845c2f963.gif)

Thank u!